### PR TITLE
Ability to change metrics setting for individual adapter

### PR DESCRIPTION
**Pull request by Keallar**
Ability to change metric settings for individual adapters

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /.bundle/
 /.yardoc
+/.idea/
```
Comment (Envek) [inline .gitignore:3]: Please don't add non-project related files to project gitignore. Set up and use global gitignore file on your machine instead. See https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer
```suggestion
```

File: lib/yabeda/metric.rb [hunk 2]
```diff
@@ -14,6 +14,7 @@ class Metric
     option :per,     optional: true, comment: "Per which unit is measured `unit`. E.g. `call` as in seconds per call"
     option :group,   optional: true, comment: "Category name for grouping metrics"
     option :aggregation, optional: true, comment: "How adapters should aggregate values from different processes"
+    option :adapter, optional: true, comment: "Which adapter apply the indicator settings."
```
Comment (Envek) [inline lib/yabeda/metric.rb:17]: Let's reword comment a bit for clarity:
```suggestion
    option :adapter, optional: true, comment: "Monitoring system adapter to register metric in and report metric values to (other adapters won't be used)"
```

File: lib/yabeda/histogram.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@ def measure(tags, value = nil)
 
       all_tags = ::Yabeda::Tags.build(tags, group)
       values[all_tags] = value
-      ::Yabeda.adapters.each do |_, adapter|
+      ::Yabeda.adapters.each_value do |adapter|
```
Comment (Envek) [inline lib/yabeda/histogram.rb:23]: Shouldn't adapters also be filtered here? (and in other metric types)

File: lib/yabeda/metric.rb
Comment (Envek) [inline lib/yabeda/metric.rb:1]: Maybe metric itself should have `adapters` method (that will default to all registered adapters unless limiting option specified)?

File: spec/yabeda_spec.rb
Comment (Envek) [inline spec/yabeda_spec.rb:1]: I believe we also need to test a case when adapter is registered, but metrics doesn't use it (so we need to have at least two adapters declared in spec file)

File: lib/yabeda/dsl/class_methods.rb [hunk 4]
```diff
@@ -115,6 +116,15 @@ def register_group_for(metric)
 
         group.register_metric(metric)
       end
+
+      def register_metric_for_adapters(metric)
```
Comment (Envek) [inline lib/yabeda/dsl/class_methods.rb:120]: I feel that this some of this logic should be in `Metric` class.

**Review by Envek — COMMENTED** [review]
Hey, thank you very much for your pull request!

Can you please tell more about your use case? What are you doing and why some metrics need to use only one adapter of many (and what are these?)

I believe we can figure out some nice API based on your insights.

Some thoughts:
 - Maybe option should be not only on metric but also on group level? So e.g. all rails/sidekiq/(your app) metrics can be limited to a single adapter?
 - Probably metric object should be responsible for calculation of the list of adapters it should be reported to.

Let's discuss!

**Comment by Keallar** [comment]
Hello! Thanks for your review!
To be honest, theres is no specific case, I just wanted to help in the development of the library that I and other uses.
About your thoughts:
1. Good idea! I'll add this in the next commit
2. I didn't quite understand this sentence. Can you please describe it more detail

File: .gitignore (hunk 1 above)
Comment (Keallar) [reply .gitignore:3]: Ok, I'll fix it

File: lib/yabeda/histogram.rb (hunk 3 above)
Comment (Keallar) [reply lib/yabeda/histogram.rb:23]: Yes, you are right! Adapters should be filtered

File: lib/yabeda/metric.rb
Comment (Keallar) [reply lib/yabeda/metric.rb:1]: I'll try to add it

File: spec/yabeda_spec.rb
Comment (Keallar) [reply spec/yabeda_spec.rb:1]: Yes, you are right

**Comment by Keallar** [comment]
@Envek, I've fixed all!

**Comment by Envek** [comment]
Hey @Keallar, thanks for improving the pull request!

I edited your pull request further a bit: moved more logic into `Yabeda::Metric` class. What do you think?

**Comment by Keallar** [comment]
Hi @Envek, yes, it looks better this way. I liked your solution to define the `adapters` method in the Metric class.
Thanks for your improving!

**Comment by Keallar** [comment]
But i need to add and fix some test cases

**Comment by Keallar** [comment]
I've done

**Review by Envek — APPROVED** [review]

**Comment by Keallar** [comment]
Thanks for cooperation!

**Comment by Envek** [comment]
Released in [0.13.0](https://github.com/yabeda-rb/yabeda/releases/tag/v0.13.0), thanks!

**Pull request merged**